CONTRIBUTING.md

```diff
@@ -52,6 +52,10 @@ We also encourage rational use of in-line variable annotation when the type of a
 
 - Numpy provides an excellent [development workflow](https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html) that we encourage you to follow when developing features for starfish!
 
+- All commits must have informative names; try to think about what will still make sense looking back on them next year.
+
+- When merging a pull request, squash commits down to the smallest logical number of commits. Use the "Rebase and Merge" or "Squash and Merge" strategy.
```

Comment:
I think squash and merge is preferable because it slaps the PR # into the commit title, by the way.

Reply:
Got it. I'll update the language for this before I merge it. I use rebase and merge when I think it's useful to have the commits remain separate longer term (My PRs tend to be bigger than yours 🙃)